pkg/endpointmanager/manager.go

```diff
@@ -652,6 +652,10 @@ type policyRepoGetter interface {
 // node's known labels.
 func (mgr *EndpointManager) InitHostEndpointLabels(ctx context.Context) {
 	ep := mgr.GetHostEndpoint()
+	if ep == nil {
+		log.Error("Attempted to init host endpoint labels but host endpoint not set.")
```

Comment:
Is it ok that we just log and continue when this happens? Seems like this might mean we want to fail fast out of the Agent?

Reply:
Depends, 

I think the original intention was for this function to get called always after checkint of GetHostEndpoint() returns 'nil' or not. But, as you may agree with, thats a bit brittle since it puts an order on function calls in the already hectic Daemon. 

For now, I just logged it. I guess, we could make it return an error if GetHostEndpoint() returns a nil. 

I wouldn't issue a Fatal from here, since its a method on EndpointManager, which I don't think should have the responsibility to kill the process all together, rather if a Daemon obj is calling it, it can make that decisions.